internal/services/privatednsresolver/private_dns_resolver_inbound_endpoint_resource.go

```diff
@@ -77,8 +77,11 @@ func (r PrivateDNSResolverInboundEndpointResource) Arguments() map[string]*plugi
 						ValidateFunc: commonids.ValidateSubnetID,
 					},
 
+					// When `private_ip_allocation_method` is `Dynamic`, API would automatically set the value for `private_ip_address`
+					// When `private_ip_allocation_method` is `Static`, `private_ip_address` needs to be set by user
```

Comment:
Could we put this as validation in the create instead of documenting it here?

Reply:
Updated

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/19754191/79f026df-494e-4fb5-b393-aa68dd816bf2)